src/Fcf/Data/List.hs

```diff
@@ -50,10 +54,49 @@ type instance Eval (Foldr f y (x ': xs)) = Eval (f x (Eval (Foldr f y xs)))
 data UnList :: b -> (a -> b -> Exp b) -> [a] -> Exp b
 type instance Eval (UnList y f xs) = Eval (Foldr f y xs)
 
+-- Helper for the Unfoldr.
+data UnfoldrCase :: (b -> Exp (Maybe (a,b))) -> Maybe (a,b) -> Exp [a]
+type instance Eval (UnfoldrCase f ('Just ab)) =
+  Eval (Fst ab) ': Eval (Unfoldr f (Eval (Snd ab)))
+type instance Eval (UnfoldrCase _ 'Nothing) = '[]
+
+-- | Type-level Unfoldr.
+--
+-- Example:
+--
+-- @
+-- data ToThree :: Nat -> Exp (Maybe (Nat,Nat))
```

Comment:
Put spaces after commas, and no padding spaces between parentheses (e.g., `(Eval (b Fcf.>= 4))` below)

Reply:
ok, made those changes to examples and the added code